internal/services/oracle/oracle_exascale_db_storage_vault_resource.go

```diff
@@ -0,0 +1,292 @@
+	"github.com/hashicorp/go-azure-sdk/resource-manager/oracledatabase/2025-03-01/exascaledbstoragevaults"
+
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+)
+
+var _ sdk.Resource = ExascaleDbStorageVaultResource{}
+
+type ExascaleDbStorageVaultResource struct{}
+
+type ExascaleDbStorageVaultResourceModel struct {
+	Location          string            `tfschema:"location"`
+	Name              string            `tfschema:"name"`
+	ResourceGroupName string            `tfschema:"resource_group_name"`
+	Tags              map[string]string `tfschema:"tags"`
+	Zones             zones.Schema      `tfschema:"zones"`
+
+	// Required
+	AdditionalFlashCacheInPercent int64                                `tfschema:"additional_flash_cache_in_percent"`
+	Description                   string                               `tfschema:"description"`
+	DisplayName                   string                               `tfschema:"display_name"`
+	HighCapacityDatabaseStorage   []ExascaleDbStorageInputDetailsModel `tfschema:"high_capacity_database_storage"`
+
+	// Optional
+	TimeZone string `tfschema:"time_zone"`
+}
+
+type ExascaleDbStorageInputDetailsModel struct {
+	TotalSizeInGb int64 `tfschema:"total_size_in_gb"`
+}
+
+func (ExascaleDbStorageVaultResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"location": commonschema.Location(),
+
+		"name": {
+			Type:     pluginsdk.TypeString,
+			Required: true,
+			ForceNew: true,
+			ValidateFunc: validation.All(
+				validation.StringLenBetween(1, 255),
+				validation.StringMatch(regexp.MustCompile(`^[a-zA-Z_]`), "Name must start with a letter or underscore (_)"),
+				validation.StringDoesNotContainAny("--"),
+			),
+		},
+
+		"resource_group_name": commonschema.ResourceGroupName(),
+
+		// Required
+		"additional_flash_cache_in_percent": {
+			Type:         pluginsdk.TypeInt,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.IntBetween(0, 100),
+		},
+
+		"description": {
+			Type:     pluginsdk.TypeString,
+			Optional: true,
+			ForceNew: true,
+		},
+
+		"display_name": {
+			Type:     pluginsdk.TypeString,
+			Required: true,
+			ForceNew: true,
+			ValidateFunc: validation.All(
+				validation.StringLenBetween(1, 255),
+				validation.StringMatch(regexp.MustCompile(`^[a-zA-Z_]`), "Display Name must start with a letter or underscore (_)"),
+				validation.StringDoesNotContainAny("--"),
+			),
+		},
+
+		"high_capacity_database_storage": {
+			Type:     pluginsdk.TypeList,
+			MinItems: 1,
+			MaxItems: 1,
+			Required: true,
+			ForceNew: true,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"total_size_in_gb": {
+						Type:     pluginsdk.TypeInt,
+						Required: true,
+						ForceNew: true,
+					},
+				},
+			},
+		},
```

Comment:
As `high_capacity_database_storage` only has one property, could it be flattened into this?
```suggestion
		"capacity_in_gb": {
			Type:     pluginsdk.TypeInt,
			Required: true,
			ForceNew: true,
		},
```